scripts/annotation-am-migration-main.sql

```diff
@@ -0,0 +1,116 @@
+
+BEGIN;
+
+DROP TABLE IF EXISTS access_management_migration_errors;
+CREATE TABLE access_management_migration_errors (
+    document_id	VARCHAR,
+    accessor_type ACCESSOR_TYPE,
+    user_id VARCHAR,
+    annotation_id VARCHAR,
+    permissions VARCHAR,
+    resource_name VARCHAR
+);
```

Comment:
The init script still needs to read the data file into a staging table in order to insert relevant data to the supporting tables... in this case it needs to insert resources into the resources table that have each resource_name given in the data file. At the moment there is only one resource_name (documentAnnotation), and if this is always going to be the case then we don't need to read the data file in this script (just insert one resource to the resources table with service_name: Annotations, resource_type: ANNOTATION, resource_name: documentAnnotation), but if there are ever going to be other resource_name values in the data file then it would be necessary to read them from the file

Reply:
Bear in mind these init scripts are only used for testing purposes, when we run the real migration the supporting tables should have the resources data inserted already

Reply:
I get what you're saying, but the fact is that the service name and resource definition are NOT already set up in some environments (eg. AAT, where the DB is cleaned to a certain point each day). The init scripts are there purely for making it easier to test the main scripts, **they will never be run during the actual migration**.

Reply:
The stage table is a temp table which means it will always be deleted once the script has finished, whether it finishes or hits an error